src/app-logic/url-handling.js

```diff
@@ -763,6 +764,30 @@ const _upgraders = {
     }
     processedLocation.query.transforms = stringifyTransforms(transforms);
   },
+  [5]: ({ query }: ProcessedLocationBeforeUpgrade) => {
+    // We changed how the ranges are serialized to the URLs. Before it was the
+    // pair of start/end in seconds unit with 4 digits, now this is a pair of
+    // start/duration, where start and duration are both integers expressed with
+    // a specific unit: for example 10000000u500 is a range from 10000000 microseconds
+    // (that is 10s) to 10s + 500 µs.
+    if (!query.range) {
+      return;
+    }
+
+    query.range = query.range
+      .split('~')
+      .map(committedRange => {
+        const m = committedRange.match(/^(-?[0-9.]+)_(-?[0-9.]+)$/);
```

Comment:
It was mostly copied from the previous implementation (see https://github.com/firefox-devtools/profiler/pull/2627/files#diff-419848f30bce72c8d9ae50b0336a7dbcL25) that was there forever ;) Except I added ^ and $ to make sure we use the whole string (I can remove them if you prefer, they may be useless because regexp modifies are greedy by default).

If I would have done it myself I would have written the code like you proposed. However I thought it was safer to use the same regexp than the previous implementation.

Reply:
BTW I would be interested to know what you find complicated here. I'd be happy to help and explain the parts that are more unclear.

Reply:
This is a very useful tool to debug regexp visually: https://www.debuggex.com/
I like especially this one, because it explains in actual english words what each part of the regexp does, but staying concise.

https://regexr.com/ is another tool that's less concise but more explicit.

Reply:
I think I can be convinced removing this regexp, even if it means this is a different implementation than previously. Tell me what you think from my explanations :)

Reply:
> It was mostly copied from the previous implementation

Ah, then I'm not concerned. I didn't realize it we've been using it awhile. As stated, since this is in an upgrader, it's write once and leave it forever.

> BTW I would be interested to know what you find complicated here. I'd be happy to help and explain the parts that are more unclear.

For me, I can take the time to read the regex and understand it. I feel like most of the time regexes don't trip me up anymore. I have a general rule of thumb to question the use of a regex anytime I see it (in my own code, and in reviews). I feel they are hard to debug, hard to read, hard to understand the expected input, and hard to maintain. So in general, if one is used, I think it should be well documented with what the expectations of the writer are, with example inputs, ideally tests for the different cases outlined, (e.g. proof that this works).

https://twitter.com/ProvablyFlarnie/status/1278717350489948161

I don't think you need to do this here, since it's an upgrader with an old regex, but in general for documenting I prefer this style:

```
// Capture two numbers, separated by a `_`.
// e.g. "-12.00_45", "123.45_-40300"
//
//                               This capture group will capture a number.
//                               |           Another number
//                               v           v
const m = committedRange.match(/^(-?[0-9.]+)_(-?[0-9.]+)$/);
```

Reply:
I added a small comment right before the regexp
```js
        // This regexp captures two (positive or negative) numbers, separated by a `_`.
```

and also augmented the comment at the start of the upgreader:
```js
    // For example we'll convert from 1.2345_2.3456 (that is: a range starting
    // at 1.2345s and ending at 2.3456s) to 1234m1112 (a range starting at
    // 1234ms and ending after 1112ms). You notice that the range is slightly
    // bigger, because this accounts for the loss of precision.
```